src/openforms/forms/models/form_variable.py

```diff
@@ -245,7 +276,29 @@ def check_data_type_and_initial_value(self):
         if self.source == FormVariableSources.user_defined:
             self.initial_value = check_initial_value(self.initial_value, self.data_type)
 
+    def check_if_prefill_plugin_mapping_needed(self):
+        """
+        This is used for user_defined variables which have the mappings for pre-filling
+        component variables. These are not aware of the prefill_plugin so we have to update them.
+        """
+
+        mappings = self.prefill_options["variables_mapping"]
+        form_variable_keys = [item["variable_key"] for item in mappings]
+        form_variables_for_update = FormVariable.objects.filter(
+            key__in=form_variable_keys
+        )
+        for variable in form_variables_for_update:
+            variable.prefill_plugin = self.prefill_plugin
+
+        FormVariable.objects.bulk_update(form_variables_for_update, ["prefill_plugin"])
```

Comment:
this seems wrong - `prefill_options` does not have a fixed structure, the shape and data inside can only be handled by the respective plugin.

I think I understand what you're trying to do here - make sure that the mapped variables get their data from the retrieved object even though no explicit prefill configuration was assigned to them. However, I think that logic belongs in the prefill plugin (objects API) implementation itself.

I now also understand the check constraint above.